### ISSUE TITLE
Cannot run a function with RedisPubSubTrigger

### ISSUE THREAD

**Issue by roman-kalivoda** [Azure/azure-functions-redis-extension#151]
Hello,

I can't get the RedisPubSubTrigger in java function to start and I can't figure out how to make it work. The function I want to run is similar to the java SimplePubSubTrigger example in the samples. I am using spring-cloud-function-adapter-azure to use spring with azure functions, but by the error I assume the problem is in initializing of the trigger.
function entry method:
```
import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.redis.annotation.RedisPubSubTrigger;
import org.springframework.stereotype.Component;

@Component
public class RedisEventListener {
  @FunctionName("SimplePubSubTrigger")
  public void run(
      @RedisPubSubTrigger(name = "req", connection = "redisConnectionString", channel = "pubsubTest", dataType = "string") String message,
      final ExecutionContext context) {
    context.getLogger().info(message);
  }
}
```
I use the preview version of extension bundle to get the extension:
```
{
  "version": "2.0",
  ...
  "extensions": {
  },
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle.Preview",
    "version": "[4.11.*, 5.0.0)"
  },
  "functions": [],
...
}
```
I have the redisConnectionString key in local.settings.json with connection string copied from the azure redis cache settings.
The generated function.json looks fine to me:
```
{
  "scriptFile" : "../ben-redis-event-trigger-1.0.0-SNAPSHOT.jar",
  "entryPoint" : "com...RedisEventListener.run",
  "bindings" : [ {
    "type" : "redisPubSubTrigger",
    "direction" : "in",
    "name" : "req",
    "channel" : "pubsubTest",
    "connectionStringSetting" : "redisConnectionString"
  } ]
}
```

but when I want to run the function it fails:
```
Azure Functions Core Tools
Core Tools Version:       4.0.5530 Commit hash: N/A +c8883e7f3c06e2b424fbac033806c19d8d91418c (64-bit)
Function Runtime Version: 4.28.5.21962

[2024-03-19T06:38:50.780Z] No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. Azure Storage, ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. builder.AddAzureStorage(), builder.AddServiceBus(), builder.AddTimers(), etc.).
[2024-03-19T06:38:50.827Z] The 'SimplePubSubTrigger' function is in error: Unable to configure binding 'req' of type 'redisPubSubTrigger'. This may indicate invalid function.json properties. Can't figure out which ctor to call.
```

EDIT: I tried installing the extension directly by removing the extensionBundle from host.json and running `func extensions install --package Microsoft.Azure.WebJobs.Extensions.Redis --version 0.5.0-preview` but I still get the same error.

**Comment by mapalan** [comment]
Apologies for the delay, but after some investigating, I found out that the `default` keyword is what is causing the issues as those `default` values aren't being populated in the generated `function.json` file. In my testing, only if a function has a value different from the `default` will the value be populated in the `function.json` file and the function correctly work. I'm currently figuring out how this was fixed in [the java annotations for other triggers](https://github.com/Azure/azure-functions-java-library/blob/dev/src/main/java/com/microsoft/azure/functions/annotation/KafkaTrigger.java) and bringing that fix here.
